Collections/collections_home_task.py

```diff
@@ -0,0 +1,458 @@
+"""
+Home task for Module 2 - Collections
+
+Write a code, which will:
+
+1. create a list of random number of dicts (from 2 to 10)
+
+dict's random numbers of keys should be letter,
+dict's values should be a number (0-100),
+example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+2. get previously generated list of dicts and create one common dict:
+
+if dicts have same key, we will take max value, and rename key with dict number with max value
+if key is only in one dict - take it as is,
+example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+Each line of code should be commented with description.
+
+Commit script to git repository and provide link as home task result.
+"""
+import random  # Import of random for generation of random numbers and letters in dictionaries
+import string  # Import of string for operations with keys of dictionaries
+
+# Create a list of random number of dicts (from 2 to 10)
+list_of_random_dictionaries = []  # Create empty list
+for i in range(0, random.randint(2, 10)):  # FOR statement with randon number of iterations between 2 and 10
+    sample_dictionary = {random.choice(string.ascii_lowercase): random.randint(0, 100)
+                         for y in range(0, random.randint(0, 100))}  # Generation of random dictionary
+    list_of_random_dictionaries.append(sample_dictionary)  # Adding of generated dictionary to list
+print(f'List of random dictionaries: {list_of_random_dictionaries}.')  # Printing of final list
+
+# Get previously generated list of dicts and create one common dict
+final_dict = {}  # Create empty final dictionary
+common_dict = {}  # Create empty temporary dictionary
+dictionary_number = 0  # Variable for storing of dictionary number
+for item in list_of_random_dictionaries:  # For each item in list with random dictionaries
+    for key, value in item.items():  # Take key and value for each dictionary
+        common_dict.update({key + f'_{dictionary_number}': value})  # Add key_dictionary_number:value to temp dictionary
+    dictionary_number = dictionary_number + 1  # Increase dictionary_number
+max_key = ""  # Empty variable for keys with maximum values
+dict_a = {}  # Set of empty temporary dictionaries for each letter separately
+temp_a = {}
+dict_b = {}
+temp_b = {}
+dict_c = {}
+temp_c = {}
+dict_d = {}
+temp_d = {}
+dict_e = {}
+temp_e = {}
+dict_f = {}
+temp_f = {}
+dict_g = {}
+temp_g = {}
+dict_h = {}
+temp_h = {}
+dict_i = {}
+temp_i = {}
+dict_j = {}
+temp_j = {}
+dict_k = {}
+temp_k = {}
+dict_l = {}
+temp_l = {}
+dict_m = {}
+temp_m = {}
+dict_n = {}
+temp_n = {}
+dict_o = {}
+temp_o = {}
+dict_p = {}
+temp_p = {}
+dict_q = {}
+temp_q = {}
+dict_r = {}
+temp_r = {}
+dict_s = {}
+temp_s = {}
+dict_t = {}
+temp_t = {}
+dict_u = {}
+temp_u = {}
+dict_v = {}
+temp_v = {}
+dict_w = {}
+temp_w = {}
+dict_x = {}
+temp_x = {}
+dict_y = {}
+temp_y = {}
+dict_z = {}
+temp_z = {}
```

Comment:
You definetely don't need to create dicts for each letter :)
Just implement a loop which will go through dicts and collect existing keys and create a new dict which will contain all these keys you've found. Of course with handling conditions of task.

Reply:
Fixed.